src/ecu/utils/helpers.ts

```
@@ -478,122 +507,50 @@ export const parseVinFromResponse = (
-    // Check if the *entire* response might be the VIN hex (unlikely but possible)
-    if (assembledHexData.length === 34) {
-      void log.warn(
-        `[Helper:parseVin] No signature found, but data length is 34 hex chars. Attempting to parse as VIN.`,
-      );
-      payloadStartIndex = 0; // Try parsing the whole string
-    } else {
-      return null; // No signature and wrong length
-    }
+  // For CAN responses, look for VIN data after service response (49 02 01)
+  const vinMatch = cleanedHex.match(/4902(01)?([0-9A-F]+)/);
+  
```

ESLint

[error] Delete `··`

(prettier/prettier)